servo/connectors/kubernetes.py

```diff
@@ -3643,13 +3653,17 @@ async def destroy(self, error: Optional[Exception] = None) -> None:
 
         self.logger.success(f'destroyed tuning Pod "{self.tuning_pod_name}"')
 
+    async def shutdown(self, error: Optional[Exception] = None) -> None:
+        # (not called - see handle_error(), defined for completeness)
+        await self.destroy(error)
+
     async def handle_error(self, error: Exception) -> bool:
-        if self.on_failure == FailureMode.rollback or self.on_failure == FailureMode.destroy:
+        if self.on_failure == FailureMode.rollback or self.on_failure == FailureMode.shutdown:
             # Ensure that we chain any underlying exceptions that may occur
             try:
                 if self.on_failure == FailureMode.rollback:
                     self.logger.warning(
-                        f"cannot rollback a tuning Pod: falling back to destroy: {error}"
+                        f"cannot rollback a tuning Pod: falling back to shutdown: {error}"
                     )
 
                 await asyncio.wait_for(self.destroy(), timeout=self.timeout.total_seconds())
```

Comment:
I think we should call shutdown here for the sake of consistency

Reply:
I thought we were leaving the 'destroy' method named 'destroy', because that's what we do for 'pod shutdown' (per your earlier comment: https://github.com/opsani/servox/pull/317#discussion_r696840440 )

Reply:
Thats correct, the method definition does not need to change but you defined a pod `shutdown` method a few lines above that does not get used that calls the `destroy` method. I'm saying we should use that method in `handle_error` for the sake of consistency with DeploymentOptimization, but the destroy method definition does not need to be changed

https://github.com/opsani/servox/blob/0deeb2d5b7af7f0f529e52b28670db84b974f173/servo/connectors/kubernetes.py#L3656

Reply:
Ok, that is an option, too - changing it. (although I defined it merely to ensure that the 'abstract' method is covered by an actual definition, not intending to use the method, actually - since handle_error() does its special thing inline).